plugins/crd/plugin_impl_crd.go

```diff
@@ -113,8 +113,40 @@ func (plugin *Plugin) Init() error {
 	return nil
 }
 
+func networkServiceValidation() *apiextv1beta1.CustomResourceValidation {
+	minLength := int64(4)
+	maxLength := int64(16)
+	validation := &apiextv1beta1.CustomResourceValidation{
+		OpenAPIV3Schema: &apiextv1beta1.JSONSchemaProps{
+			Properties: map[string]apiextv1beta1.JSONSchemaProps{
+				"spec": apiextv1beta1.JSONSchemaProps{
+					Required: []string{"name"},
+					Properties: map[string]apiextv1beta1.JSONSchemaProps{
+						"name": apiextv1beta1.JSONSchemaProps{
+							Type:        "string",
+							MinLength:   &minLength,
+							MaxLength:   &maxLength,
+							Description: "NetworkService Name",
+							Pattern:     `^[a-zA-Z0-9]+\-[a-zA-Z0-9]*$`,
+						},
+					},
+				},
+			},
+		},
+	}
+	return validation
+}
+
 // Create the CRD resource, ignore error if it already exists
 func createCRD(plugin *Plugin, FullName, Group, Version, Plural, Name string) error {
+
+	var validation *apiextv1beta1.CustomResourceValidation
```

Comment:
Since this function is generic, maybe we can pass the function definition in, or assign based on the resource type or name passed in.

Reply:
For now I am using `switch` for `name` to select corresponding validator function. Would it be ok to continue using it?

Reply:
👍